### DISCUSSION TITLE
Clarification regarding possible changes for input file sorting in Incremental Clustering

### DISCUSSION THREAD

**Discussion by Anurag19101996** [uw-ssec/MAWpy#41] [General]
Currently get_preprocessed_dataframe (a function in the utilities folder) is sorting the input dataframe by user id and by time, sequentially, and this function is called at the beginning of incremental clustering. However, sorting by time might not be necessary at the beginning (Wang and Chen’s 2019 paper did not specify this step). 

@qzchen-uw @gracejia513 @anujsinha3 @carlosgjs

**Comment by anujsinha3** [comment]
If you see the old code, All operations are performed on sorted days

https://github.com/uw-ssec/MAWpy/blob/main/src/mawpy/IncrementalClustering.py#L241
https://github.com/uw-ssec/MAWpy/blob/main/src/mawpy/IncrementalClustering.py#L409
https://github.com/uw-ssec/MAWpy/blob/main/src/mawpy/IncrementalClustering.py#L441

I assume detecting stays from contiguous temporal/spatial information only makes sense if the traces for a user are in sorted order.
Hence, the data frame has been sorted by time.

**Comment by qzchen-uw** [comment]
@anujsinha3 I assume that clustering itself did not use the time information?
